### ISSUE TITLE
[BUG] A couple of test_csvreader.py tests prints out CUDA errors

### ISSUE THREAD

**Issue by j-ieong** [rapidsai/cudf#707]
**Description**
The `test_csvreader.py::test_csv_reader_nrows` and `test_csvreader.py::test_csv_reader_skiprows_header[2-4]` tests print out errors despite the test itself passing.

**Repro**
1. Run one of the above tests with `-s` and `-v` to show printf output and verbose, respectively:
```console
pytest -k 'test_csv_reader_nrows' -sv
```
2. Output shows errors but marked as passed. Expect no errors to be printed:
```python
======================================================= test session starts ========================================================
platform linux -- Python 3.5.5, pytest-3.8.1, py-1.7.0, pluggy-0.8.1 -- /home/jieong/miniconda3/envs/cudf_dev/bin/python
cachedir: .pytest_cache
rootdir: /home/jieong/cudf/python, inifile:
collected 39 items / 38 deselected                                                                                                 

cudf/tests/test_csvreader.py::test_csv_reader_nrows ERROR: CUDA Runtime call cudaGetLastError() in line 1261 of file /home/jieong/cudf/cpp/src/io/csv/csv_reader.cu failed with invalid configuration argument (9).
Empty DataFrame
Columns: ['int1', 'int2']
Index: []
PASSED
```

**Environment**
 - Ubuntu 18 LTS
 - cuDF Conda install
 - branch-0.5 TOT

**Comment by vuule** [comment]
Please assign this one to me.
Looks like the issue is specific to the case where the resulting dataframe is empty.

**Comment by j-ieong** [comment]
@mjsamoht To assign, since I don't have permissions

**Comment by kkraus14** [comment]
Resolved by #717
